### PR TITLE
Corrected the message to work with None values in world_axis_physical_types

### PR DESCRIPTION
**Pull request by yashrsharma44**
This PR addresses the issue of displaying correct message of wcs when one/more values of `world_axis_physical_types` contains `None` values.
## Previous Issue
When I run commands as - 
```
from astropy.wcs.wcsapi.sliced_low_level_wcs import SlicedLowLevelWCS
wcs1 = SlicedLowLevelWCS(wt, [])
wcs1
```
I get the following
Error - 
```
~/ndcub/vnv/lib/python3.7/site-packages/astropy-4.0.dev24714-py3.7-linux-x86_64.egg/astropy/wcs/wcsapi/low_level_api.py in __str__(self)
    313         # Find largest between header size and value length
    314         world_dim_width = max(9, len(str(self.world_n_dim)))
--> 315         world_typ_width = max(13, max(len(x) for x in self.world_axis_physical_types))
    316 
    317         s += (('{0:' + str(world_dim_width) + 's}').format('World Dim') + '  ' +

~/ndcub/vnv/lib/python3.7/site-packages/astropy-4.0.dev24714-py3.7-linux-x86_64.egg/astropy/wcs/wcsapi/low_level_api.py in <genexpr>(.0)
    313         # Find largest between header size and value length
    314         world_dim_width = max(9, len(str(self.world_n_dim)))
--> 315         world_typ_width = max(13, max(len(x) for x in self.world_axis_physical_types))
    316 
    317         s += (('{0:' + str(world_dim_width) + 's}').format('World Dim') + '  ' +

TypeError: object of type 'NoneType' has no len()

```
My WCS object contains a `world_physical_value` as `None` type.

I think @Cadair can provide more context to this issue. 

## Current Output

```
SlicedLowLevelWCS Transformation

This transformation has 4 pixel and 4 world dimensions

Array shape (Numpy order): (1, 2, 3, 4)

Pixel Dim  Data size  Bounds
        0          4  None
        1          3  None
        2          2  None
        3          1  None

World Dim  Physical Type                   Units
        0  time                            min
        1  em.wl                           m
        2  custom:pos.helioprojective.lat  deg
        3  None                            unknown

Correlation between pixel and world axes:

               Pixel Dim
World Dim    0    1    2    3
        0  yes   no   no   no
        1   no  yes   no   no
        2   no   no  yes  yes
        3   no   no  yes  yes

```

### CODE REVIEW

**Comment by Cadair** [comment]
Elements of the `world_axis_physical_types` list being none is allowed by APE 14, so this is indeed a bug. It would probably be good if you could add a regression test. @yashrsharma44 :smile: 

ping @astrofrog

**Comment by yashrsharma44** [comment]
Ping @bsipocz 
Shall I add a changelog for this bug fix?

**Comment by bsipocz** [comment]
yes, bug fixes usually need entries in the changelog. Put in in the 3.2.1 `wcs` section please.

**Comment by Cadair** [comment]
This has also broken the SunPy example gallery :man_facepalming:

File: CHANGES.rst [hunk 1]
```diff
@@ -322,6 +322,13 @@ astropy.visualization
 astropy.wcs
 ^^^^^^^^^^^
 
+- Do not attempt to delete repeated distortion keywords multiple times when
```
Comment (Cadair) [inline CHANGES.rst]: I assume this isn't supposed to be here?

Comment (yashrsharma44) [reply CHANGES.rst]: I actually rebased my branch, and it merged this change as well. Any suggestions to remove it using git, or shall I remove it manually?

Comment (bsipocz) [reply CHANGES.rst]: @yashrsharma44 - the rebase should never bring in a merge commit. I would suggest to rebase again. When you're done you need to force push your branch back to the remote rather that pull/merge.

Comment (yashrsharma44) [reply CHANGES.rst]: Actually, first I updated my `master` branch using `git pull upstream master` `upstream` being my remote to astropy repo. Then I performed `git rebase none-fix`, so it included the merge commit in the rebase. 
Now I have reset the merge, and tried doing rebase again, but it still catches those extra lines. May be I need to see which commit catches the extra lines. Thanks for the insight of never having a merge commit in a rebase, I will keep that in my mind.

Comment (yashrsharma44) [reply CHANGES.rst]: Now its correct, and I am getting a conflict error.

Comment (yashrsharma44) [reply CHANGES.rst]: @bsipocz Can you help me with the conflicting files?

Comment (pllim) [reply CHANGES.rst]:
```
git fetch upstream master
git rebase upstream/master
```

Follow screen instructions to resolve conflict in editor.

```
git add CHANGES.rst
git rebase --continue
```

Comment (yashrsharma44) [reply CHANGES.rst]: Thanks @pllim for the suggestions, it worked!

**Comment by nden** [comment]
Looks reasonable.

**Comment by bsipocz** [comment]
@astrofrog @pllim - the changelog section inconsistency for this PR has tricked both the bot and the release scripts. I clean this up from master, but the release has been sent out with the wrong changelog. Not sure why it happened.

**Comment by pllim** [comment]
Might be a bug...

https://github.com/astropy/astropy-bot/blob/7982bb95f58bfef5961dc6642c8697f5cedac389/astropy_bot/changelog_checker.py#L59-L61

**Pull request merged**